### ISSUE TITLE
refactor: Remove redundant View castings in Services

### ISSUE THREAD

**Issue by srv-twry** [opendatakit/opendatakit#1388]
#### Software and hardware versions 
Latest code from development branch.

#### Problem description
Currently there are a lot of instances of redundant View castings in the project. 
For eg. 
``` Java
infoPane = (LinearLayout) view.findViewById(R.id.sync_info_pane);
```
can be replaced by 
```Java
infoPane = view.findViewById(R.id.sync_info_pane);
```
They were required before but aren't required now and Android studio suggests removing them.

#### Steps to reproduce the problem
NA

#### Expected behavior
They should be removed.

#### Other information 
NA

@jbeorse Views ? 
Quick win for first timers may be :smile:

**Comment by jbeorse** [comment]
Oh that is interesting. Does this reduce the amount of lint warnings we will get?

**Comment by srv-twry** [comment]
@jbeorse 
Yes i think so. Code inspection shows 65 instances of Redundant type casts in the project. A large number of those are these ```findViewById``` castings.
[Here](https://stackoverflow.com/questions/44902651/no-need-to-cast-the-result-of-findviewbyid) is a stackoverflow discussion on this.

**Comment by jbeorse** [comment]
Our compileVersion is 27, so it does seem like this should work. Good find! I'll mark it as a quick win, certainly contributor friendly. Go ahead and claim it if you like. 😄

**Comment by jbeorse** [comment]
Your title mentions only Services, but I imagine this is true in Survey, Tables, and Scan as well. I think we should extend it to all 4 apps.

**Comment by srv-twry** [comment]
@opendatakit-bot claim

**Comment by linl33** [comment]
Hi @srv-twry thank you for your pull requests!

There are some more redundant casts in [androidlibrary](https://github.com/opendatakit/androidlibrary), after those are fixed I will close this issue.
